Firestore/core/src/firebase/firestore/model/transform_operations.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_MODEL_TRANSFORM_OPERATIONS_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_MODEL_TRANSFORM_OPERATIONS_H_
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+// TODO(zxu123): We might want to refactor transform_operations.h into several
+// files when the number of different types of operations grows gigantically.
+// For now, put the interface and the only operation here.
+
+/** Represents a transform within a TransformMutation. */
+class TransformOperation {
+ public:
+  /** All the different kinds to TransformOperation. */
+  enum class Type {
+    ServerTimestamp,
+  };
+
+  /** Returns the actual type. */
+  virtual Type type() const = 0;
+
+  /** Returns whether the two are equal. */
+  virtual bool operator==(const TransformOperation* other) const = 0;
```

Comment:
Optional: I'd avoid making a `virtual` operator, though it's hard to give concrete suggestions without knowing how `TransformOperation` is expected to evolve.

Reply:
ack. Here a virtual method is required. I can change it to, say `equals()`.